src/server/snapshot.cc

```diff
@@ -273,23 +273,21 @@ void SliceSnapshot::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req)
 }
 
 void SliceSnapshot::OnJournalEntry(const journal::Entry& entry) {
-  CHECK(journal::Op::VAL == entry.opcode);
-
   optional<RdbSerializer> tmp_serializer;
   RdbSerializer* serializer_ptr = default_serializer_.get();
-  if (entry.db_ind != current_db_) {
+  if (entry.dbid != current_db_) {
     CompressionMode compression_mode = compression_mode_ == CompressionMode::NONE
                                            ? CompressionMode::NONE
                                            : CompressionMode::SINGLE_ENTRY;
     tmp_serializer.emplace(compression_mode);
     serializer_ptr = &*tmp_serializer;
   }
 
-  PrimeKey pkey{entry.key};
-  SerializeEntry(entry.db_ind, pkey, *entry.pval_ptr, entry.expire_ms, serializer_ptr);
+  CHECK(entry.opcode == journal::Op::COMMAND);
+  serializer_ptr->WriteJournalEntries(absl::Span{&entry, 1});
 
   if (tmp_serializer) {
-    FlushTmpSerializer(entry.db_ind, &*tmp_serializer);
+    FlushTmpSerializer(entry.dbid, &*tmp_serializer);
```

Comment:
I am trying to understand the flow of the journal flush to sink.
Only in stable state this function OnJournalEntry is called, right?
If we have only one db index in the server than we wont use the tmp_serializer. Where FlushDefaultBuffer will be called?

Reply:
ok great, so I believe this things should be writen to the design doc.

Reply:
Its not needed, we only care about the relative order

We don't have to send the changes continuously because as long as the replica is in loading state, it doesn't process any queries

Reply:
Keeping a large buffer is indeed a problem